### PR TITLE
Kubernetes[run_workloads]: adapt run_workloads to support kubernetes.

### PR DESCRIPTION
**Pull request by Damenus**

### CODE REVIEW

**Comment by felidadae** [comment]
+3400 lines?
You need to make some merge/rebase. Do You started your work from allocation branch? Currently is hard to look at the PR code from github. Please fix this.

File: workloads/redis/redis.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import os
+from common import *
+
+# ----------------------------------------------------------------------------------------------------
+###
+# Params which can be modified by exporting environment variables.
+###
+
+# Port that stressed application listens on.
```
Comment (felidadae) [inline workloads/redis/redis.py]: remove this and next two lines

File: workloads/redis/pod.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+apiVersion: v1
```
Comment (felidadae) [inline workloads/redis/pod.yaml]: Drop file - I suppose that it is not needed anymore.

File: workloads/__init__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2018 Intel Corporation
```
Comment (felidadae) [inline workloads/__init__.py]: Dropb file

File: workloads/common.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    'load_generator': load_generator,
+    'job_uniq_id': job_uniq_id,
+    'env_uniq_id': env_uniq_id,
+    'own_ip': own_ip,
+    'application_host_ip': application_host_ip,
+    'load_generator_host_ip': load_generator_host_ip,
+}
+wrapper_labels.update(extra_labels)
+
+# For debug purposes.
```
Comment (felidadae) [inline workloads/common.py]: drop, commented

File: workloads/common.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    def pre_create_job(self, config):
+        for label_nama, label_value in self.labels.items():
+            config.add_metadata(label_nama, label_value)
+        return config
+
+
+def dedent(s):
+    return textwrap.dedent(s).replace('\n',' ').replace('\'', '"')
+
+# WorkloadService = Service(
```
Comment (felidadae) [inline workloads/common.py]: drop

File: workloads/common.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        "containers": containers,
+        #"affinity": affinity
+        "nodeSelector": {
+            "own_ip": own_ip
+        }
+    }
+}
+
+
+#hooks = [AddMetadata(wrapper_labels)]
```
Comment (felidadae) [inline workloads/common.py]: remove

File: workloads/common.py [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+    }
+]
+
+metadata = {
+    "name": job_name.replace('_','-').replace('.','-'),
+    "labels": wrapper_labels
+}
+
+affinity = {
```
Comment (felidadae) [inline workloads/common.py]: please change formatting - it should take 10 lines and it takes 20

File: workloads/common.py [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+pod = {
+    "apiVersion": "v1",
+    "kind": "Pod",
+    "metadata": metadata,
+    "spec": {
+        "volumes": volumes,
+        "hostNetwork": True,
+        "initContainers": initContainers,
+        "containers": containers,
+        #"affinity": affinity
```
Comment (felidadae) [inline workloads/common.py]: why commented?

File: workloads/redis/redis.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+
+communication_port = os.environ.get('communication_port', 11211)
+# ----------------------------------------------------------------------------------------------------
+
+
+vloume = {"name": "shared-data"}
+#volumes.append(vloume)
```
Comment (felidadae) [inline workloads/redis/redis.py]: why commented?

File: workloads/redis/redis.py [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+cmdline = "redis-server redis.conf"
+cmd = "redis-server /prep_config/redis.conf"
+command.append(cmd)
+
+json_format = json.dumps(pod)
+print(json_format)
+
+
+# jobs = [
```
Comment (felidadae) [inline workloads/redis/redis.py]: please drop

File: workloads/redis/redis1.yaml [hunk 11]
```diff
@@ -0,0 +1,52 @@
+apiVersion: v1
```
Comment (felidadae) [inline workloads/redis/redis1.yaml]: please drop file

File: workloads/run_workloads_inventory.yaml [hunk 12]
```diff
@@ -0,0 +1,160 @@
+# This is a template inventory for run_workloads.yaml ansible playbook.
```
Comment (felidadae) [inline workloads/run_workloads_inventory.yaml]: Please drop the file, just write in the PR description what's needed to set in template file to make it work - also add in the PR information about the most recent DOCKER_TAG build (if that is important)

File: workloads/redis/redis.py (hunk 1 above)
Comment (Damenus) [reply workloads/redis/redis.py]: done

File: workloads/redis/pod.yaml (hunk 2 above)
Comment (Damenus) [reply workloads/redis/pod.yaml]: Done

File: workloads/__init__.py (hunk 3 above)
Comment (Damenus) [reply workloads/__init__.py]: Done

File: workloads/common.py (hunk 4 above)
Comment (Damenus) [reply workloads/common.py]: Done

File: workloads/common.py (hunk 5 above)
Comment (Damenus) [reply workloads/common.py]: Done

File: workloads/common.py (hunk 6 above)
Comment (Damenus) [reply workloads/common.py]: Done

File: workloads/common.py (hunk 7 above)
Comment (Damenus) [reply workloads/common.py]: I've removed affinity. At the end I used nodeSelector

File: workloads/common.py (hunk 8 above)
Comment (Damenus) [reply workloads/common.py]: Now it is removed

File: workloads/redis/redis.py (hunk 9 above)
Comment (Damenus) [reply workloads/redis/redis.py]: Both lines not required. I've removed it

File: workloads/redis/redis.py (hunk 10 above)
Comment (Damenus) [reply workloads/redis/redis.py]: Done

File: workloads/redis/redis1.yaml (hunk 11 above)
Comment (Damenus) [reply workloads/redis/redis1.yaml]: Done

File: workloads/rpc_perf/rpc_perf.yaml [hunk 13]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
```
Comment (ppalucki) [inline workloads/rpc_perf/rpc_perf.yaml]: drop it

Comment (Damenus) [reply workloads/rpc_perf/rpc_perf.yaml]: done

File: workloads/run_workloads_inventory.yaml (hunk 12 above)
Comment (Damenus) [reply workloads/run_workloads_inventory.yaml]: file removed

**Pull request merged**